lighthouse-core/gather/gather-runner.js

```diff
@@ -115,6 +115,24 @@ class GatherRunner {
     return {};
   }
 
+  /**
+   * Reject if the gathering terminates due to the CDP target crashing, etc
+   * @param {Driver} driver
+   * @return {Promise<void>}
+   */
+  static async getGatherTerminatedPromise(driver) {
+    return new Promise((_, reject) => {
+      driver.on('Inspector.targetCrashed', _ => {
```

Comment:
if iframes crash should we still consider it fatal?

Reply:
good question.. if an iframe shares a process with its parent.. it takes down both. (naturally)

but here's a iframe page crash when that iframe is isolated (oopif):

![image](https://user-images.githubusercontent.com/39191/102440967-1ef73200-3fd6-11eb-98b1-12a0779234f2.png)

and.......... 

if we didn't explicitly attach to that iframe target.. we'd just see a `Target.targetCrashed` event, but not a `Inspector.targetCrashed` one.

but we DO  autoattach to iframe targets. so we're attached. and thus we'll get the full fledged `Inspector.targetCrashed`.  :)



i don't think there's an easy call here, but ... probably yes.

Reply:
nice sleuthing thanks :) 

I suppose if we ever have a single gatherer that tries to evaluate something in subtarget contexts we will need to support this (framehole maybe?), so SGTM 👍